### PR TITLE
[IOTDB-2867] Response leader redirect when the current ConfigNode is not leader 

### PR DESCRIPTION
**Pull request by CRZbulabula**
The ConfigNode only accept RPC requests when it's the leader of the ConfigNodeGroup. And return TSStatusCode.NEED_REDIRECTION otherwise.

result

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java [hunk 1]
```diff
@@ -139,24 +149,44 @@ public DataSet getOrCreateSchemaPartition(PhysicalPlan physicalPlan) {
...
+  }
+
+  private TSStatus confirmLeader() {
+    Endpoint endpoint = getConsensusManager().getLeader();
+    if (endpoint == null) {
+      return new TSStatus(TSStatusCode.NEED_REDIRECTION.getStatusCode())
+          .setMessage(
+              "The current ConfigNode is not leader. And ConfigNodeGroup is in leader election. Please redirect with a random ConfigNode.");
+    }
+    if (Objects.equals(
```
[the remark this answers is not in the document]
Comment (wangchao316) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java:181]: yes, first use isLeader(). second if is not leader, use getLeder.

File: confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/server/ConfigNodeRPCServerProcessor.java [hunk 2]
```diff
@@ -82,12 +77,21 @@ public TDataNodeRegisterResp registerDataNode(TDataNodeRegisterReq req) throws T
...
+    if (dataSet.getStatus().getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+      LOGGER.info(
+          "Register DataNode successful. DataNodeId: {}, Endpoint: {}",
+          dataSet.getDataNodeId(),
+          req.getEndPoint().toString());
+    } else {
+      LOGGER.info(
+          "Register DataNode {} failed. {}",
+          req.getEndPoint().toString(),
+          dataSet.getStatus().getMessage());
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/server/ConfigNodeRPCServerProcessor.java:90]: remove log print.  print log when execute logical.

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java]: fixed~

**Review by SzyWilliam — APPROVED** [review]
LGTM

File: confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/server/ConfigNodeRPCServerProcessor.java (hunk 2 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/server/ConfigNodeRPCServerProcessor.java]: fixed.

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**